ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -305,28 +392,25 @@ localRootPeersProvider tracer timeout resolvConf rootPeersVar domains = do
             withResource' (TraceLocalRootFailure domain `contramap` tracer)
                           (1 :| [3, 6, 9, 12])
                           rr
-          reply <- lookupAWithTTL timeout resolvConf resolver daDomain
+          reply <- resolveDomain resolver domain advertisePeer
           case reply of
-            Left  err -> do
-              traceWith tracer (TraceLocalRootFailure domain (DNSError err))
-              go rrNext (ttlForDnsError err ttl)
-
+            Left err -> go rrNext (ttlForDnsError err ttl)
             Right results -> do
-              traceWith tracer (TraceLocalRootResult domain results)
               atomically $ do
-                rootPeers <- readTVar rootPeersVar
-                let resultsMap :: Map Socket.SockAddr PeerAdvertise
-                    resultsMap = Map.fromList [ ( Socket.SockAddrInet
-                                                    daPortNumber
-                                                    (IP.toHostAddress addr)
-                                                , advertisePeer)
-                                              | (addr, _ttl) <- results ]
-                    rootPeers' :: Map DomainAddress (Map Socket.SockAddr PeerAdvertise)
-                    rootPeers' = Map.insert domain resultsMap rootPeers
+                rootPeersGroups <- readTVar rootPeersGroupsVar
+                let rootPeersGroups' = IntMap.fromList $ zip [0,1..] rootPeersGroups
+                    (target, entry)  = rootPeersGroups' IntMap.! group
+                    resultsMap       = Map.fromList (map fst results)
+                    newEntry         = Map.union entry resultsMap
```

Comment:
```suggestion
                    entry'         = entry
                                    <> Map.fromList (map fst results)
```

* `rootPeersGroups'` denote the new value, so let's be consistent and use `entry'` here as well.
* maybe it's just me, but using `<>` makes it more obvious that we're doing the right thing.

Reply:
Agree!